src/utils/collapseSourcemaps.ts

```diff
@@ -49,23 +48,20 @@ class Link {
 		const sourcesContent: string[] = [];
 		const names: string[] = [];
 
-		const mappings = this.mappings.map(line => {
+		const mappings = [];
+
+		for (const line of this.mappings) {
 			const tracedLine: SourceMapSegmentVector[] = [];
```

Comment:
And possibly here as well, although it may not make that much of a difference, but at least avoids possible reallocation cost.

Reply:
In my experience building computational geometry libraries that process arrays with millions of items, it's safe to initialize an empty array without worrying about performance. The only case where you want (and forced to) to preallocate is with typed arrays.

I also prefer `for ... of` over `forEach`/`map` because it avoids new closures. TypeScript converts them into simple `for` loops.

Reply:
> TypeScript converts them into simple for loops

Ah right, I stumbled about this once but forgot later. Very nice, so there is no performance overhead checking the array iterator.